From 2706270287c1743caa6b18462fd206de9b0b15aa Mon Sep 17 00:00:00 2001
From: Hari Naraayana Desikan Kannan <quic_hnaraaya@quicinc.com>
Date: Sun, 3 Nov 2024 12:12:56 +0530
Subject: [PATCH] OpenSSL-3.0.10 patch update

Patch the openssl package with version 3.0.10 with below updates.
1. Select libopenssl-legacy if openssl-util is selected.
2. Load legacy provider by default in openssl.cnf.

Change-Id: I1c48389acf76ad5b808f22b866fb239791ea2b09
Signed-off-by: Hari Naraayana Desikan Kannan <quic_hnaraaya@quicinc.com>
(cherry picked from commit f173b2d43e93073ee5f987c3312e1b47053e4395)
---
 package/libs/openssl/Makefile                             | 2 +-
 .../patches/150-openssl.cnf-add-engines-conf.patch        | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/package/libs/openssl/Makefile b/package/libs/openssl/Makefile
index 70d64d67116d..5537eab4d0c1 100644
--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -108,7 +108,7 @@ define Package/openssl-util
   $(call Package/openssl/Default)
   SECTION:=utils
   CATEGORY:=Utilities
-  DEPENDS:=+libopenssl +libopenssl-conf
+  DEPENDS:=+libopenssl +libopenssl-conf +libopenssl-legacy
   TITLE+= (utility)
 endef
 
diff --git a/package/libs/openssl/patches/150-openssl.cnf-add-engines-conf.patch b/package/libs/openssl/patches/150-openssl.cnf-add-engines-conf.patch
index 9fe9cdf590cd..1b1653e37116 100644
--- a/package/libs/openssl/patches/150-openssl.cnf-add-engines-conf.patch
+++ b/package/libs/openssl/patches/150-openssl.cnf-add-engines-conf.patch
@@ -10,7 +10,7 @@ Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>
 
 --- a/apps/openssl.cnf
 +++ b/apps/openssl.cnf
-@@ -52,10 +52,13 @@ tsa_policy3 = 1.2.3.4.5.7
+@@ -52,10 +52,14 @@ tsa_policy3 = 1.2.3.4.5.7
  
  [openssl_init]
  providers = provider_sect
@@ -19,18 +19,22 @@ Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>
  # List of providers to load
  [provider_sect]
  default = default_sect
++legacy = legacy_sect
 +.include /var/etc/ssl/providers.cnf
 +
  # The fips section name should match the section name inside the
  # included fipsmodule.cnf.
  # fips = fips_sect
-@@ -69,7 +72,13 @@ default = default_sect
+@@ -69,7 +73,16 @@ default = default_sect
  # OpenSSL may not work correctly which could lead to significant system
  # problems including inability to remotely access the system.
  [default_sect]
 -# activate = 1
 +activate = 1
 +
++[legacy_sect]
++activate = 1
++
 +[engines_sect]
 +.include /var/etc/ssl/engines.cnf
 +
-- 
2.34.1

